app/Controllers/authController.php

```diff
@@ -187,8 +187,8 @@ public function formLoginAction(): void {
 				Minz_Request::forward(['c' => 'auth', 'a' => 'login'], false);
 			}
 		} elseif (FreshRSS_Context::systemConf()->unsafe_autologin_enabled) {
-			$username = Minz_Request::paramString('u');
-			$password = Minz_Request::paramString('p');
+			$username = Minz_Request::paramString('u', specialchars: true);
+			$password = Minz_Request::paramString('p', specialchars: true);
```

Comment:
This works, but it's confusing.

Elsewhere it says something intuitive like:

https://github.com/FreshRSS/FreshRSS/blob/31c8846791d4b5316fbc790202f79545c012f9c2/lib/Minz/Request.php#L78

But the guilty function reverses that:

https://github.com/FreshRSS/FreshRSS/blob/31c8846791d4b5316fbc790202f79545c012f9c2/lib/Minz/Request.php#L114

Reply:
I am not sure to understand what you mean, @Frenzie .
And `paramArrayString()` is not used here

Reply:
paramArrayString and all the other old functions say

```
if specialchars 
  return htmlspecialchars($result)
else
  result $result
```

This new addition `paramStringNull` says:
```
if specialchars
  return $result
else
  return htmlspecialchars($result)
```

I find this very unexpected, and it's the reason I didn't see where the problem came from at a glance.

Reply:
Right, I see what you mean, and that is indeed unfortunate, and should be made uniform.
The original function is the following one:

https://github.com/FreshRSS/FreshRSS/blob/e5320759ebe5bd820e2f815265ecd289f899842f/lib/Minz/Request.php#L45-L58

Reply:
But let's address than in another PR

Reply:
https://github.com/FreshRSS/FreshRSS/pull/6800